eventsources/sources/bitbucketserver/custombitbucketserverclient.go

```diff
@@ -0,0 +1,117 @@
+package bitbucketserver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// customBitbucketServerClient returns a Bitbucket Server HTTP client that implements methods that gfleury/go-bitbucket-v1 does not.
+// Specifically getting Pull Requests associated to a commit is not supported by gfleury/go-bitbucket-v1.
+type customBitbucketServerClient struct {
+	client *http.Client
+	ctx    context.Context
+	token  string
+	url    *url.URL
+}
+
+// pullRequestRes is a struct containing information about the Pull Request.
+type pullRequestRes struct {
+	ID    int    `json:"id"`
+	State string `json:"state"`
+}
+
+// pagedPullRequestsRes is a paged response with values of pullRequestRes.
+type pagedPullRequestsRes struct {
+	Size          int              `json:"size"`
+	Limit         int              `json:"limit"`
+	IsLastPage    bool             `json:"isLastPage"`
+	Values        []pullRequestRes `json:"values"`
+	Start         int              `json:"start"`
+	NextPageStart int              `json:"nextPageStart"`
+}
+
+type pagination struct {
+	Start int
+	Limit int
+}
+
+func (p *pagination) StartStr() string {
+	return strconv.Itoa(p.Start)
+}
+
+func (p *pagination) LimitStr() string {
+	return strconv.Itoa(p.Limit)
+}
+
+func (c *customBitbucketServerClient) authHeader(req *http.Request) {
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.token))
```

Comment:
Some BBservers are only configured for Basic authentication even with the use of PA tokens. Is it possible change the Header set based on the flag under the accessToken section in https://github.com/argoproj/argo-events/blob/master/examples/event-sources/bitbucketserver.yaml#L37

Reply:
This change is big enough we should open a new PR for that.